core/scheduler/service/executor_input_compiler.go

```diff
@@ -62,6 +63,33 @@ type InputCompiler struct {
 	logger log.Logger
 }
 
+func sanitiseLabel(key string) string {
+	// implements validation from https://cloud.google.com/bigquery/docs/labels-intro#requirements
+
+	// max length 63 chars
+	// if length greater than 63 then truncate to 61 and add __ at end to denote truncation
+	if maxLabelLength := 63; len(key) > maxLabelLength {
+		key = "__" + key[len(key)-61:]
+	}
+	// Keys can contain only lowercase letters, numeric characters, underscores, and dashes.
+	key = strings.ToLower(key)
+	re := regexp.MustCompile(`[^\w-]`)
+	key = re.ReplaceAllString(key, "-")
+	return key
+}
+
+func getJobLabelsString(labels map[string]string) string {
+	labelString := ""
+	for key, value := range labels {
+		var separator string
+		if len(labelString) != 0 {
+			separator = ","
+		}
+		labelString = labelString + separator + fmt.Sprintf("%s=%s", sanitiseLabel(key), sanitiseLabel(value))
```

Comment:
just a small thing. why don't we use join instead? example:

```go
var sanitizedLabels []string

// some iteration ...
sanitizedLabels = append(sanitizedLabels, fmt.Sprintf ...)
// outside the iteration ...

strings.Join(sanitizedLabels, separator)
```

this way, we don't need to remember the separator thing in each iteration

Reply:
sure